core/autoscaling_test.go

```diff
@@ -2808,3 +2970,64 @@ func TestGetAllowedInstaceTypes(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPricetoBid(t *testing.T) {
+	tests := []struct {
+		multiplier           float64
+		currentSpotPrice     float64
+		currentOnDemandPrice float64
+		flag                 bool
+		want                 float64
+	}{
+		{
+			multiplier:           1.0,
+			currentSpotPrice:     0.0216,
+			currentOnDemandPrice: 0.0464,
+			flag:                 true,
+			want:                 0.0216,
+		},
+		{
+			multiplier:           1.5,
+			currentSpotPrice:     0.0216,
+			currentOnDemandPrice: 0.0464,
+			flag:                 true,
+			want:                 0.0324,
+		},
+		{
+			multiplier:           1.79,
+			currentSpotPrice:     0.0216,
+			currentOnDemandPrice: 0.0464,
+			flag:                 true,
+			want:                 0.038664,
+		},
+		{
+			multiplier:           1.79,
+			currentSpotPrice:     0.0216,
+			currentOnDemandPrice: 0.0464,
+			flag:                 false,
+			want:                 0.0464,
+		},
+	}
+	for _, tt := range tests {
+		cfg := &Config{
+			SpotPriceMultiplier: tt.multiplier,
+		}
+		asg := &autoScalingGroup{
+			name: "testSpotPriceMultiplier",
+			region: &region{
+				name: "us-east-1",
+				conf: cfg,
+			},
+			SpotPriceMultiplierFlag: tt.flag,
+		}
+		// asg.region.determineInstanceTypeInformation(cfg)
+		// newInstance := asg.region.instanceTypeInformation["t2.medium"]
+		currentSpotPrice := tt.currentSpotPrice
+		currentOnDemandPrice := tt.currentOnDemandPrice
+		actualPrice := asg.getPricetoBid(currentOnDemandPrice, currentSpotPrice)
+		if math.Abs(actualPrice-tt.want) > 0.000001 {
+			t.Errorf("multiplier = %.2f, flag = %t, expected price = %.5f, want %.5f, currentSpotPrice = %.5f",
+				tt.multiplier, tt.flag, actualPrice, tt.want, currentSpotPrice)
+		}
+	}
+}
```

Comment:
Tests not yet reviewed due to previous comments, please let me know if you'd like me to do so anyway